File: sources/Android/android-communications/library/src/main/java/com/polar/androidcommunications/api/ble/model/polar/PolarAdvDataUtility.kt
package com.polar.androidcommunications.api.ble.model.polar

object PolarAdvDataUtility {

    fun getDeviceModelNameFromAdvLocalName(advLocalName: String, withPrefixToTrim: String = "Polar"): String {
        return if (isValidDevice(advLocalName, withPrefixToTrim)) {
            val modelName = advLocalName.trim()
                .replaceFirst((if (withPrefixToTrim != "") "$withPrefixToTrim " else ""), "")
            val endIndex = modelName.lastIndexOf(" ")
            modelName.substring(0, endIndex)
        } else {
            ""
        }
    }

    fun isValidDevice(name: String, requiredPrefix: String = "Polar"): Boolean {
        return name.trim().startsWith(requiredPrefix) && name.trim()
            .split(" ").size > 2
    }
}

File: sources/iOS/ios-communications/Sources/iOSCommunications/ble/api/model/polar/PolarAdvDataUtility.swift
import Foundation

open class PolarAdvDataUtility {
        
    public static func getDeviceNameFromAdvLocalName(advLocalName: String, withPrefixToTrim prefix: String = "Polar") -> String {
        if (isValidDevice(advLocalName: advLocalName, requiredPrefix: prefix)) {
            let modelName = advLocalName.trimmingCharacters(in: .whitespacesAndNewlines)
                .replacingOccurrences(of: prefix != "" ? prefix + " " : "", with: "")
            
            if let endIndex = modelName.lastIndex(of: " ") {
                let mySubstring = modelName[..<(endIndex)]
                return String(mySubstring)
            } else {
                return ""
            }
        } else {
            return ""
        }
    }
    
    public static func isValidDevice(advLocalName: String, requiredPrefix: String = "Polar") -> Bool {
        return advLocalName.trimmingCharacters(in: .whitespacesAndNewlines).hasPrefix(requiredPrefix) &&
            advLocalName.trimmingCharacters(in: .whitespacesAndNewlines).split(separator: " ").count > 2
    }
}
